tests/time_tracker_api/users/users_namespace_test.py

```diff
@@ -6,20 +6,68 @@
 from pytest import mark
 
 
-def test_users_response_contains_expected_props(
-    client: FlaskClient, valid_header: dict,
+@patch(
+    'commons.feature_toggles.feature_toggle_manager.FeatureToggleManager.get_azure_app_configuration_client'
+)
+@patch(
+    'commons.feature_toggles.feature_toggle_manager.FeatureToggleManager.is_toggle_enabled_for_user'
+)
+@patch('utils.azure_users.AzureConnection.users')
+@patch('utils.azure_users.AzureConnection.users_v2')
+def test_feature_toggle_is_on_then_role_field_is_list(
+    users_v2_mock,
+    users_mock,
+    is_toggle_enabled_for_user_mock,
+    get_azure_app_configuration_client_mock,
+    client: FlaskClient,
+    valid_header: dict,
 ):
 
-    AzureConnection.users = Mock(
-        return_value=[{'name': 'dummy', 'email': 'dummy', 'role': 'dummy'}]
-    )
+    is_toggle_enabled_for_user_mock.return_value = True
+    users_v2_mock.return_value = [
+        {'name': 'dummy', 'email': 'dummy', 'roles': ['dummy-role']}
+    ]
+    response = client.get('/users', headers=valid_header)
+
+    users_v2_mock.assert_called()
+    users_mock.assert_not_called()
+    assert HTTPStatus.OK == response.status_code
+    assert 'name' in json.loads(response.data)[0]
+    assert 'email' in json.loads(response.data)[0]
+    assert 'roles' in json.loads(response.data)[0]
+    assert ['dummy-role'] == json.loads(response.data)[0]['roles']
 
-    response = client.get('/users', headers=valid_header,)
 
+@patch(
+    'commons.feature_toggles.feature_toggle_manager.FeatureToggleManager.get_azure_app_configuration_client'
+)
+@patch(
+    'commons.feature_toggles.feature_toggle_manager.FeatureToggleManager.is_toggle_enabled_for_user'
+)
+@patch('utils.azure_users.AzureConnection.users')
+@patch('utils.azure_users.AzureConnection.users_v2')
+def test_feature_toggle_is_off_then_role_field_is_string(
```

Comment:
Is it possible to simplify the test to execute the two functions in one, since the code is almost the same in the two functions? Is it possible to parameterize the mocking of the users? It's just a suggestion.

Reply:
Yes the problem to parametrize is this code:
```
Test 1
users_mock.assert_called()  
users_v2_mock.assert_not_called()

Test 2
users_mock.assert_not_called()
users_v2.assert_called()
``` 
But, I did not find a convenient way to parametrize that.